tools/cli/commands/creategpu.py

```diff
@@ -31,79 +31,121 @@
-download/v1.0.1/nvidia-docker_1.0.1-1_amd64.deb
-    dpkg -i ./nvidia-docker_1.0.1-1_amd64.deb
-    apt-get update -y
-    apt-get install nvidia-docker -y
-  fi
-}}
-
-cleanup_packages() {{
-  apt-get update -y
-  apt-get remove -y dnsmasq-base || true
-}}
-
-pull_datalab_image() {{
-  if [[ "$(docker images -q {0})" == "" ]]; then
-    gcloud docker -- pull {0} ;
-  fi
-}}
-
-start_datalab_docker() {{
-  nvidia-docker run --restart always -p '127.0.0.1:8080:8080' \
-    -e DATALAB_ENV='GCE' -e DATALAB_DEBUG='true' \
-    -e DATALAB_SETTINGS_OVERRIDES=\
-'{{"enableAutoGCSBackups": {3}, "consoleLogLevel": "{4}" }}' \
-    -e DATALAB_GIT_AUTHOR='{5}' \
-    -v "${{MOUNT_DIR}}/content:/content" \
-    -v "${{MOUNT_DIR}}/tmp:/tmp" \
-    {0} -c /datalab/run.sh
-}}
-
-start_fluentd_docker() {{
-  docker run --restart always \
-    -e FLUENTD_ARGS='-q' \
-    -v /var/log:/var/log \
-    -v /var/lib/docker/containers:/var/lib/docker/containers:ro \
-    gcr.io/google_containers/fluentd-gcp:1.18
-}}
-
-cleanup_packages
-install_cuda
-install_nvidia_docker
-cleanup_packages
-pull_datalab_image
-mount_and_prepare_disk
-configure_swap
-cleanup_tmp
-start_datalab_docker
-start_fluentd_docker
-
-journalctl -u google-startup-scripts --no-pager > /var/log/startupscript.log
+    NVidia GPU Driver: NVIDIA-Linux-x86_64-384.81""")
+
+# The config for the 'cos-gpu-installer.service'
+# services comes from the 'GoogleCloudPlatform/cos-gpu-installer' project
+# here: https://github.com/GoogleCloudPlatform/cos-gpu-installer
+
+_DATALAB_CLOUD_CONFIG = """
+#cloud-config
+
+users:
+- name: datalab
+  uid: 2000
+  groups: docker
+- name: logger
+  uid: 2001
+  groups: docker
+
+write_files:
+- path: /etc/nvidia-installer-env
+  permissions: 0755
+  owner: root
+  content: |
+    NVIDIA_DRIVER_VERSION=384.81
+    COS_NVIDIA_INSTALLER_CONTAINER=gcr.io/cos-cloud/cos-gpu-installer:latest
+    NVIDIA_INSTALL_DIR_HOST=/var/lib/nvidia
+    NVIDIA_INSTALL_DIR_CONTAINER=/usr/local/nvidia
+    ROOT_MOUNT_DIR=/root
+
+- path: /etc/systemd/system/cos-gpu-installer.service
+  permissions: 0755
+  owner: root
+  content: |
+    [Unit]
+    Description=Run the GPU driver installer container
+    Requires=network-online.target gcr-online.target
+    After=network-online.target gcr-online.target
+
+    [Service]
+    User=root
+    Type=oneshot
+    RemainAfterExit=true
+    EnvironmentFile=/etc/nvidia-installer-env
+    ExecStartPre=/bin/bash -c 'mkdir -p "${{NVIDIA_INSTALL_DIR_HOST}}" && mount --bind "${{NVIDIA_INSTALL_DIR_HOST}}" "${{NVIDIA_INSTALL_DIR_HOST}}" && mount -o remount,exec "${{NVIDIA_INSTALL_DIR_HOST}}"'
+    ExecStart=/usr/bin/docker run --privileged --net=host --pid=host --volume "${{NVIDIA_INSTALL_DIR_HOST}}":"${{NVIDIA_INSTALL_DIR_CONTAINER}}" --volume /dev:/dev --volume "/":"${{ROOT_MOUNT_DIR}}" --env-file /etc/nvidia-installer-env "${{COS_NVIDIA_INSTALLER_CONTAINER}}"
+    ExecStartPost=/bin/bash -c "${{NVIDIA_INSTALL_DIR_HOST}}/bin/nvidia-smi"
```

Comment:
Why do we need to run this?  Is this just so the GPU info is in the logs?

Reply:
Removed.